api/restHandler/AppListingRestHandler.go

```diff
@@ -144,23 +144,58 @@ func (handler AppListingRestHandlerImpl) FetchAppsByEnvironment(w http.ResponseW
 	t2 := time.Now()
 	handler.logger.Infow("api response time testing", "time", time.Now().String(), "time diff", t2.Unix()-t1.Unix(), "stage", "2")
 	t1 = t2
-	appEnvs := make([]*bean.AppEnvironmentContainer, 0)
-	rbacObjects := handler.enforcerUtil.GetRbacObjectsForAllApps()
-	for _, env := range envContainers {
-		if fetchAppListingRequest.DeploymentGroupId > 0 {
-			if env.EnvironmentId != 0 && env.EnvironmentId != dg.EnvironmentId {
-				continue
+
+	isActionUserSuperAdmin, err := handler.userService.IsSuperAdmin(int(userId))
+	if err != nil {
+		handler.logger.Errorw("request err, FetchAppsByEnvironment", "err", err, "userId", userId)
+		writeJsonResp(w, err, "Failed to check is super admin", http.StatusInternalServerError)
+		return
+	}
+	appEnvContainers := make([]*bean.AppEnvironmentContainer, 0)
+	if isActionUserSuperAdmin {
+		for _, envContainer := range envContainers {
+			appEnvContainers = append(appEnvContainers, envContainer)
+		}
+	} else {
+		var teamIds []*int
+		authorizedTeams := make(map[int]int)
+		for _, envContainer := range envContainers {
+			teamIds = append(teamIds, &envContainer.TeamId)
```

Comment:
should we not get all unique teams and make query to FetchAppsByEnvironment after that only for valid teams, this way the query will be faster

Reply:
done